pixelgl/shader.go

```diff
@@ -0,0 +1,162 @@
+package pixelgl
+
+import (
+	"github.com/faiface/glhf"
+	"github.com/faiface/mainthread"
+	"github.com/go-gl/mathgl/mgl32"
+	"github.com/pkg/errors"
+)
+
+// GLShader is a type to assist with managing a canvas's underlying
+// shader configuration. This allows for customization of shaders on
+// a per canvas basis.
+type (
```

Comment:
use one 'type' keyword per type here, because the second type is not exported, so it's not needed to couple it with an exported type